client/python/tests/integration/test_no_auth.py

```diff
@@ -172,6 +173,23 @@ def test_submit_job_and_cancel_by_queue_job_set(client: ArmadaClient, queue_name
     assert expected == cancelled_message.cancelled_ids[0]
 
 
+def test_submit_job_and_cancelling_with_filter(client: ArmadaClient, queue_name):
+    job_set_name = f"set-{uuid.uuid1()}"
```

Comment:
You almost never want a v1 UUID. This should be updated to use a UUID v4 -- especially since you're not filling in the inputs. See: https://docs.python.org/3/library/uuid.html#uuid.uuid4

```suggestion
    job_set_name = f"set-{uuid.uuid4()}"
```